exec/cnex/global.c

```diff
@@ -97,6 +107,27 @@ void print(TExecutor *exec)
     pop(exec->stack);
 }
 
+void chr(struct tagTExecutor *exec)
+{
+    Number x = top(exec->stack)->number; pop(exec->stack);
+
+    if (!number_is_integer(x)) {
+        exec->rtl_raise(exec, "ValueRangeException", "chr() argument not an integer", BID_ZERO);
+        return;
+    }
+    if (number_is_negative(x) || number_is_greater(x, number_from_uint32(0x10ffff))) {
+        exec->rtl_raise(exec, "ValueRangeException", "chr() argument out of range 0-0x10ffff", BID_ZERO);
+        return;
+    }
+
+    Cell *r = cell_createStringCell(0);
+    // ToDo: Implement UTF8 strings here!!
+    char ch = (char)number_to_uint32(x);
+    string_appendChars(r->string, &ch, 1);
```

Comment:
There are a lot of instances where you call `string_appendChars(s, &c, 1)` and only append one character. Rather than this inconvenient construction, suggest making a new `string_appendChar(TString *, char)` function that only appends *one* character. That way you can avoid needing a temporary variable like in this case, and the interface becomes simpler.

Reply:
Changed string_appendChars() to string_appendChar().  and it only takes a char, without a count.